docs/sources/guides/dataframe.md

```diff
@@ -0,0 +1,157 @@
+
+#### Time Series (Shared time values)
+
+If all the series share the same time values, then a "wide" format can be used:
+
+```text
+Name: Wide
+Dimensions: 3 Fields by 2 Rows
++---------------------+-----------------+-----------------+
+| Name: time          | Name: cpu       | Name: cpu       |
+| Labels:             | Labels: host=a  | Labels: host=b  |
+| Type: []time.Time   | Type: []float64 | Type: []float64 |
++---------------------+-----------------+-----------------+
+| 2020-01-02 03:04:00 | 3               | 4               |
+| 2020-01-02 03:05:00 | 6               | 7               |
++---------------------+-----------------+-----------------+
+```
+
+#### Time Series (Long Format)
+
+(Note: Currently supported on backend only: [Grafana Issue #22219](https://github.com/grafana/grafana/issues/22219)).
+
+A common CSV or SQL format is the [Long (a.k.a Tall/Narrow) Format](https://en.wikipedia.org/wiki/Wide_and_narrow_data) of time series data.
+
+This format is supported and is detected when there are also string columns in the dataframe. There can be multiple number and multiple string columns, and the series will be grouped.
+
+For example a Long format Series:
+
+```text
+Name: Long
+Dimensions: 4 Fields by 4 Rows
++---------------------+-----------------+-----------------+----------------+
+| Name: time          | Name: aMetric   | Name: bMetric   | Name: host     |
+| Labels:             | Labels:         | Labels:         | Labels:        |
+| Type: []time.Time   | Type: []float64 | Type: []float64 | Type: []string |
++---------------------+-----------------+-----------------+----------------+
+| 2020-01-02 03:04:00 | 2               | 10              | foo            |
+| 2020-01-02 03:04:00 | 5               | 15              | bar            |
+| 2020-01-02 03:05:00 | 3               | 11              | foo            |
+| 2020-01-02 03:05:00 | 6               | 16              | bar            |
++---------------------+-----------------+-----------------+----------------+
+```
+
+Would look like the following in "Wide" format:
+
+```text
+Name: Wide
+Dimensions: 5 Fields by 2 Rows
++---------------------+------------------+------------------+------------------+------------------+
+| Name: time          | Name: aMetric    | Name: bMetric    | Name: aMetric    | Name: bMetric    |
+| Labels:             | Labels: host=foo | Labels: host=foo | Labels: host=bar | Labels: host=bar |
+| Type: []time.Time   | Type: []float64  | Type: []float64  | Type: []float64  | Type: []float64  |
++---------------------+------------------+------------------+------------------+------------------+
+| 2020-01-02 03:04:00 | 2                | 10               | 5                | 15               |
+| 2020-01-02 03:05:00 | 3                | 11               | 6                | 16               |
++---------------------+------------------+------------------+------------------+------------------+
+```
+
+## Technical References
```

Comment:
Please add introductory paragraphs after this heading. Per the style guide, we avoid stacked headings.